### DISCUSSION TITLE
Simplify syntax

### DISCUSSION THREAD

**Discussion by jzwar** [tataratat/splinepy#163] [Ideas]
I wanted to collect some ideas at a central point, on how we can simplify the syntax, e.g., using more default arguments:
- [x] Proximity: 
      ```python
      para_cords = bspline.proximities(np.random(..., ...))
      ```
      Returning (as a default) parametric coordinates only and setting resolutions, iterations automatically

Maybe we can find some more ideas here @j042 @danielwolff1

**Comment by j042** [comment]
sounds good!  something like `return_verbose`, `return_newton_info`, (or some nice name) ...  flag and iff true return the whole thing.

**Reply by jzwar** [reply]
Would it make sense to set the default number of points to the len(ukvs) or more specifically, use greville abscissae as a basis for kd?

**Reply by j042** [reply]
greville abscissae sounds good, but would require some changes in core. Maybe we can start with some default number for now? Some other options like specifying starting point could also be nice

**Reply by jzwar** [reply]
I used the control-mesh resolution, as a temporary fix
